keylime/enrich.py

```diff
@@ -0,0 +1,289 @@
+#
+# These function use efivar C libraries to decode device path and guid
+#
+##################################################################################
+
+efivarlib = "libefivar.so"  # formerly "/usr/lib/x86_64-linux-gnu/libefivar.so"
+
+efivarlib_functions = CDLL(efivarlib)
+
+
+def getDevicePath(b):
+    ret = efivarlib_functions.efidp_format_device_path(0, 0, b, len(b))
+    if ret < 0:
+        raise Exception(
+            f'getDevicePath: efidp_format_device_path({b}) returned {ret}')
+
+    s = create_string_buffer((ret+1))
+
+    ret = efivarlib_functions.efidp_format_device_path(s, ret+1, b, len(b))
+    if ret < 0:
+        raise Exception(
+            f'getDevicePath: efidp_format_device_path({b}) returned {ret}')
+
+    return s.value.decode("utf-8")
+
+
+def getGUID(b):
+    s = c_char_p(None)
+    ret = efivarlib_functions.efi_guid_to_str(b, byref(s))
+    if ret < 0:
+        raise Exception(f'getGUID: efi_guid_to_str({b}) returned {ret}')
+    return s.value.decode("utf-8")
+
+##################################################################################
+#
+# https://uefi.org/sites/default/files/resources/UEFI%20Spec%202.8B%20May%202020.pdf
+# Section 32.4.1
+#
+# Relevant data structures
+#
+#  typedef struct _EFI_SIGNATURE_LIST {
+#       uint8_t SignatureType[16];
+#       uint32_t SignatureListSize;
+#       uint32_t SignatureHeaderSize;
+#       uint32_t SignatureSize;
+#       // uint8_t SignatureHeader[SignatureHeaderSize];
+#       // uint8_t Signatures[][SignatureSize];
+#  } EFI_SIGNATURE_LIST;
+#
+#  typedef struct _EFI_SIGNATURE_DATA {
+#      uint8_t SignatureOwner[16];
+#      uint8_t SignatureData[];
+#  } EFI_SIGNATURE_DATA;
+#
+##################################################################################
+
+##################################################################################
+# Print out EFI_SIGNATURE_DATA
+##################################################################################
+
+
+def getKey(b, start, size):
+    key = {}
+    signatureOwner = getGUID(b[start:start+16])
+    key['SignatureOwner'] = signatureOwner
+
+    signatureData = b[start+16:start+size]
+    key['SignatureData'] = signatureData.hex()
+    return key
+
+
+##################################################################################
+#
+# https://uefi.org/sites/default/files/resources/UEFI%20Spec%202.8B%20May%202020.pdf
+# Section 3.1.3
+#
+# Relevant data structures
+#
+#   typedef struct _EFI_LOAD_OPTION {
+#       UINT32 Attributes;
+#       UINT16 FilePathListLength;
+#       // CHAR16 Description[];
+#       // EFI_DEVICE_PATH_PROTOCOL FilePathList[];
+#       // UINT8 OptionalData[];} EFI_LOAD_OPTION;
+#   } EFI_LOAD_OPTION;
+#
+##################################################################################
+
+##################################################################################
+# Print additional information about variables BootOrder and Boot####
```

Comment:
Again, misleading comment since this it's not printing anything

Reply:
fixed